code/datums/components/style/style.dm

```diff
@@ -0,0 +1,499 @@
+
+/datum/component/style/proc/format_rank_string(new_rank)
+	var/rank_string = rank_to_string(new_rank)
+	var/final_string = ""
+	final_string += "<span class='maptext' style='font-size: 8px'><font color='[rank_to_color(new_rank)]'><b>[rank_string[1]]</b>"
+	final_string += "<span style='font-size: 7px'>[copytext(rank_string, 2)]</span></font></span>"
+	return final_string
+
+/datum/component/style/proc/generate_multiplier()
+	return "<br><span class='maptext' style='font-size: 7px'>MULTIPLIER: [point_multiplier]X</span>"
+
+/datum/component/style/proc/generate_actions()
+	var/action_string = ""
+	for(var/action in actions)
+		action_string += "<br><span class='maptext'>+ <font color='[action_to_color(actions[action])]'>[actions[action]]</font></span>"
+	return action_string
+
+/datum/component/style/proc/action_to_color(action)
+	switch(action)
+		if(ACTION_KILL)
+			return "#ff0000"
+		if(ACTION_MINOR_KILL)
+			return "#ff6666"
+		if(ACTION_MAJOR_KILL)
+			return "#ffaa00"
+		if(ACTION_DISRESPECT)
+			return "#990000"
+		if(ACTION_MELEED)
+			return "#660033"
+		if(ACTION_ROCK_MINED)
+			return "#664433"
+		if(ACTION_ORE_MINED)
+			return "#663366"
+		if(ACTION_TRAPPER)
+			return "#363366"
+		if(ACTION_PARRIED)
+			return "#591324"
+		if(ACTION_PROJECTILE_BOOST)
+			return "#80112c"
+		if(ACTION_GIBTONITE_HIT)
+			return "#201d40"
+		if(ACTION_GIBTONITE_BOOM)
+			return "#1e176e"
+		if(ACTION_GIBTONITE_DEFUSED)
+			return "#2b2573"
+		if(ACTION_MARK_DETONATED)
+			return "#ac870e"
+		if(ACTION_GEYSER_MARKED)
+			return "#364866"
+
+/// A proc that lets a user, when their rank >= `hotswap_rank`, swap items in storage with what's in their hands, simply by clicking on the stored item with a held item
+/datum/component/style/proc/hotswap(mob/living/source, atom/target, obj/item/weapon, proximity_flag, click_parameters)
+	SIGNAL_HANDLER
+
+	if((rank < hotswap_rank) || !isitem(target) || !(target in source.get_all_contents()))
+		return
+
+	var/obj/item/item_target = target
+
+	if(!(item_target.item_flags & IN_STORAGE))
+		return
+
+	var/datum/storage/atom_storage = item_target.loc.atom_storage
+
+	if(!atom_storage.can_insert(weapon, source, messages = FALSE))
+		source.balloon_alert(source, "unable to hotswap!")
+		return
+
+	atom_storage.attempt_insert(weapon, source, override = TRUE)
+	INVOKE_ASYNC(source, TYPE_PROC_REF(/mob/living, put_in_hands), target)
+	source.visible_message(span_notice("[source] quickly swaps [weapon] out with [target]!"), span_notice("You quickly swap [weapon] with [target]."))
+
+
+/datum/component/style/proc/on_parent_multitool(datum/source, mob/living/user, obj/item/tool, list/recipes)
+	multitooled = !multitooled
+	user.balloon_alert(user, "meter [multitooled ? "" : "un"]hacked")
+
+
+// Point givers
+/datum/component/style/proc/on_punch(mob/living/carbon/human/punching_person, atom/attacked_atom, proximity)
+	SIGNAL_HANDLER
+
+	if(!proximity || !punching_person.combat_mode || !isliving(attacked_atom))
```

Comment:
combat_mode has been replaced with istate
for non combat mode it would be !(punching_person.istate & ISTATE_HARM)

Reply:
fixed